### PR TITLE
initial import from the original repo

### PR DESCRIPTION
**Pull request by codekitchen**
I did some rebasing here on the assumption that you'll want to preserve the git history.

### CODE REVIEW

**Comment by suyograo** [comment]
@codekitchen I don't think we should be checking in client jars to GitHub repo. These are required only at build time -- when the Gem is created, so we should fetch them at gem creation time. For example, this is how we handle Kafka jars, which is similar to this library. We use a library called jar-dependencies which fetches the jar directly from maven.

See:
https://github.com/logstash-plugins/logstash-input-kafka/blob/master/logstash-input-kafka.gemspec#L22-L25

https://github.com/logstash-plugins/logstash-input-kafka/blob/master/Rakefile

`rake install_jars` is called when gem is built

If you would like, I can make the change as well, but we should definitely remove the jars from this PR

**Comment by karmi** [comment]
Hi @codekitchen, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/logstash-plugins/logstash-input-kinesis/pull/1.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by codekitchen** [comment]
Sure @suyograo that sounds reasonable -- this was my first jruby project, so I just followed what seemed to be best practice for other logstash plugins back in March 2015 when I started out. But I'm sure things have changed a lot since then.

To be clear though, you say "we should definitely remove the jars from this PR" -- the jars were already committed to the project, the only way to remove them would be to remove the git history and start fresh (or do some invasive rewriting). Do you want me to do that? Doesn't seem worth the loss of information. Or are you just asking me to create a new commit and append it to the PR? That I can do.

**Comment by suyograo** [comment]
@codekitchen `779e6ec` works. We need one more file to make `jar_dependencies` work. https://github.com/logstash-plugins/logstash-input-kafka/blob/master/lib/logstash-input-kafka_jars.rb. Can you copy contents as is under `lib/`, but call it `logstash-input-kinesis_jars.rb`

**Comment by codekitchen** [comment]
OK there we go... I haven't had a chance to test out the re-built gem yet, but the specs pass and the contents of the gem look correct, including all the jars.

File: logstash-input-kinesis.gemspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |spec|
   spec.homepage      = "https://github.com/codekitchen/logstash-input-kinesis"
   spec.licenses      = ['Apache License (2.0)']
 
-  spec.files         = `git ls-files`.split($\)
+  spec.files         = `git ls-files`.split($\) + Dir['vendor/jar-dependencies/runtime-jars/*.jar']
```
Comment (suyograo) [inline logstash-input-kinesis.gemspec]: Actually, can we change this to: 
```
s.files = Dir['lib/**/*','spec/**/*','vendor/**/*','*.gemspec','*.md','CONTRIBUTORS','Gemfile','LICENSE','NOTICE.TXT']
```

Comment (suyograo) [inline logstash-input-kinesis.gemspec]: using `git ls` was a hack for us, which didn't work well for offline installs.

File: logstash-input-kinesis.gemspec [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  spec.name          = "logstash-input-kinesis"
+  spec.version       = Logstash::Input::Kinesis::VERSION
+  spec.authors       = ["Brian Palmer"]
+  spec.email         = ["brian@codekitchen.net"]
+  spec.summary       = %q{Logstash plugin for Kinesis input}
+  spec.description   = %q{This gem is a logstash plugin required to be installed on top of the Logstash core pipeline using $LS_HOME/bin/plugin install gemname. This gem is not a stand-alone program}
+  spec.homepage      = "https://github.com/codekitchen/logstash-input-kinesis"
+  spec.licenses      = ['Apache License (2.0)']
+
+  spec.files         = `git ls-files`.split($\) + Dir['vendor/jar-dependencies/runtime-jars/*.jar']
```
Comment (suyograo) [inline logstash-input-kinesis.gemspec]: Can you change this to:

```
s.files = Dir['lib/**/*','spec/**/*','vendor/**/*','*.gemspec','*.md','CONTRIBUTORS','Gemfile','LICENSE','NOTICE.TXT']
```

Using `git ls` was a hack for us, which didn't work well for offline installs.

**Comment by suyograo** [comment]
@codekitchen one small comment about `git ls` and we can merge this.

**Comment by codekitchen** [comment]
👍 done

**Comment by suyograo** [comment]
LGTM

**Pull request merged**